core/node/libp2p/transport.go

```diff
@@ -64,6 +66,10 @@ func Transports(tptConfig config.Transports) interface{} {
 			opts.Opts = append(opts.Opts, libp2p.Transport(webrtc.New))
 		}
 
+		if tcpEnabled || wsEnabled {
+			opts.Opts = append(opts.Opts, libp2p.ShareTCPListener())
```

Comment:
Feels odd we need to do this for TCP, but not for UDP. 
Dropped comment in https://github.com/libp2p/go-libp2p/pull/2984/files#r1823606432